sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -714,6 +714,17 @@ object TypeCoercion {
         // try to find the first one we can implicitly cast.
         case (_, TypeCollection(types)) => types.flatMap(implicitCast(e, _)).headOption.orNull
 
+        case (ArrayType(_, nullable), ArrayType(internalType: DataType, expectedNullable))
+          if (expectedNullable || nullable == expectedNullable) =>
```

Comment:
Ah, actually @hvanhovell, in case of `case _ if expectedType.acceptsType(inType) => inType` would this return `None` instead of the same input type? 

I see this is being used by `implicitCast(in, expected).getOrElse(in)` so it would be fine to work but I am worried of this (e.g. for unit test and etc).